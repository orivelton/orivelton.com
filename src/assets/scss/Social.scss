@font-face {
  font-family: 'icomoon';
  src:  url('../fonts/icomoon.eot?m3y14y');
  src:  url('../fonts/icomoon.eot?m3y14y#iefix') format('embedded-opentype'),
    url('../fonts/icomoon.ttf?m3y14y') format('truetype'),
    url('../fonts/icomoon.woff?m3y14y') format('woff'),
    url('../fonts/icomoon.svg?m3y14y#icomoon') format('svg');
  font-weight: normal;
  font-style: normal;
  font-display: block;
}

[class^="icon-"], [class*=" icon-"] {
  /* use !important to prevent issues with browser extensions that change fonts */
  font-family: 'icomoon' !important;
  speak: none;
  font-style: normal;
  font-weight: normal;
  font-variant: normal;
  text-transform: none;
  line-height: 1;

  /* Better Font Rendering =========== */
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.social {
  display: flex;
  width: 7rem;
  flex-direction: row;
  justify-content: space-around;
  margin-top: 1rem;
}

.icon {
  color: #000000;
  font-size: 1.5rem;
  &__name {
    font-size: 0;
  }
}

.icon-twitter:before {
  content: "\e900";
}
.icon-github:before {
  content: "\e901";
}
.icon-linkedin:before {
  content: "\e902";
}
