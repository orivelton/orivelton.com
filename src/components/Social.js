import React from 'react';
import '../assets/scss/Social.scss';

const Social = () => {
  const socialNames = [
    {
      name: 'Linkedin',
      url: 'https://www.linkedin.com/in/orivelton/',
      icon: 'linkedin'
    },
    {
      name: 'GitHub',
      url: 'https://github.com/orivelton',
      icon: 'github'
    },
    {
      name: 'Twitter',
      url: 'https://twitter.com/orivelton_cesar',
      icon: 'twitter'
    }
  ];
  
  return (
    <ul className="social">
      {
        socialNames.map(({ name, url, icon }) => (
          <li key={name}>
            <a href={url} className={`icon icon-${icon}`} target="_blank" title={`Got to ${name}`} rel="noopener noreferrer">
              <span className="icon__name">{ name }</span>
            </a>
          </li>
        ))
      }
    </ul>
  )
};

export default Social;
