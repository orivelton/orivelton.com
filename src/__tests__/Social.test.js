import React from 'react';
import renderer from 'react-test-renderer';
import Social from '../components/Social';

describe('src/components/Social.js', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Social />)
      .toJSON()
    expect(tree).toMatchSnapshot();
  });
});
