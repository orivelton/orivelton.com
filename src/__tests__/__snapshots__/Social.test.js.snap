// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/Social.js renders correctly 1`] = `
<ul
  className="social"
>
  <li>
    <a
      className="icon icon-linkedin"
      href="https://www.linkedin.com/in/orivelton/"
      rel="noopener noreferrer"
      target="_blank"
      title="Got to Linkedin"
    >
      <span
        className="icon__name"
      >
        Linkedin
      </span>
    </a>
  </li>
  <li>
    <a
      className="icon icon-github"
      href="https://github.com/orivelton"
      rel="noopener noreferrer"
      target="_blank"
      title="Got to GitHub"
    >
      <span
        className="icon__name"
      >
        GitHub
      </span>
    </a>
  </li>
  <li>
    <a
      className="icon icon-twitter"
      href="https://twitter.com/orivelton_cesar"
      rel="noopener noreferrer"
      target="_blank"
      title="Got to Twitter"
    >
      <span
        className="icon__name"
      >
        Twitter
      </span>
    </a>
  </li>
</ul>
`;
